### ISSUE TITLE
failed to resolve broker hostname

### ISSUE THREAD

**Issue by mihalby** [ansible/awx#6477]
##### ISSUE TYPE
 - Bug Report

##### ENVIRONMENT
* AWX version: :9.3.0
* AWX install method: docker on linux
* Ansible version:   2.9.6
* Operating System: centos 7
* Web Browser: chrome

##### STEPS TO REPRODUCE

Clean first install from git repo.
 ansible-playbook -i inventory install.yml
Go to web interface and run any managemet job.

##### EXPECTED RESULTS
job is run and done

##### ACTUAL RESULTS

job state is pending

##### ADDITIONAL INFORMATION
task container logs: docker-compose logs task

awx_task     | 2020-03-30 11:57:49,942 WARNING  awx.main.commands.run_callback_receiver scaling up worker pid:175
awx_task     | 2020-03-30 11:57:49,942 WARNING  awx.main.commands.run_callback_receiver scaling up worker pid:175
awx_task     | 2020-03-30 11:57:49,942 WARNING  awx.main.dispatch scaling up worker pid:174
awx_task     | 2020-03-30 11:57:49,950 WARNING  awx.main.dispatch scaling up worker pid:176
awx_task     | 2020-03-30 11:57:49,959 WARNING  kombu.mixins Broker connection error, trying again in 2.0 seconds: OSError('failed to resolve broker hostname',).
awx_task     | Traceback (most recent call last):
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/amqp/transport.py", line 137, in _connect
awx_task     |     host, port, family, socket.SOCK_STREAM, SOL_TCP)
awx_task     |   File "/usr/lib64/python3.6/socket.py", line 745, in getaddrinfo
awx_task     |     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
awx_task     | socket.gaierror: [Errno -2] Name or service not known
awx_task     |
awx_task     | During handling of the above exception, another exception occurred:
awx_task     |
awx_task     | Traceback (most recent call last):
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/kombu/utils/functional.py", line 344, in retry_over_time
awx_task     |     return fun(*args, **kwargs)
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/kombu/connection.py", line 283, in connect
awx_task     |     return self.connection
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/kombu/connection.py", line 839, in connection
awx_task     |     self._connection = self._establish_connection()
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/kombu/connection.py", line 794, in _establish_connection
awx_task     |     conn = self.transport.establish_connection()
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/kombu/transport/pyamqp.py", line 130, in establish_connection
awx_task     |     conn.connect()
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/amqp/connection.py", line 311, in connect
awx_task     |     self.transport.connect()
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/amqp/transport.py", line 77, in connect
awx_task     |     self._connect(self.host, self.port, self.connect_timeout)
awx_task     |   File "/var/lib/awx/venv/awx/lib/python3.6/site-packages/amqp/transport.py", line 148, in _connect
awx_task     |     "failed to resolve broker hostname"))
awx_task     | OSError: failed to resolve broker hostname

**Comment by ryanpetrello** [comment]
@mihalby,

Can you share:

`docker ps -a`?

**Comment by mihalby** [comment]
docker ps
CONTAINER ID        IMAGE                                   COMMAND                  CREATED             STATUS              PORTS                                                                                    NAMES
6c81bf3e25d8        ansible/awx_task:9.3.0                  "/tini -- /bin/sh -c…"   36 seconds ago      Up 26 seconds       8052/tcp                                                                                 awx_task
17ecd11f8746        ansible/awx_web:9.3.0                   "/tini -- /bin/sh -c…"   37 seconds ago      Up 26 seconds       0.0.0.0:8081->8052/tcp                                                                   awx_web
93006e0aa62b        redis                                   "docker-entrypoint.s…"   37 seconds ago      Up 27 seconds       6379/tcp                                                                                 awx_redis
2a26d2655542        memcached:alpine                        "docker-entrypoint.s…"   37 seconds ago      Up 27 seconds       11211/tcp                                                                                awx_memcached
ee76bd7a7655        postgres:10                             "docker-entrypoint.s…"   37 seconds ago      Up 27 seconds       5432/tcp                                                                                 awx_postgres

**Comment by mihalby** [comment]
> @mihalby,
> 
> Can you share:
> 
> `docker ps -a`?

```
 docker ps
CONTAINER ID        IMAGE                                   COMMAND                  CREATED             STATUS              PORTS                                                                                    NAMES
6c81bf3e25d8        ansible/awx_task:9.3.0                  "/tini -- /bin/sh -c…"   5 minutes ago       Up 5 minutes        8052/tcp                                                                                 awx_task
17ecd11f8746        ansible/awx_web:9.3.0                   "/tini -- /bin/sh -c…"   5 minutes ago       Up 5 minutes        0.0.0.0:8081->8052/tcp                                                                   awx_web
93006e0aa62b        redis                                   "docker-entrypoint.s…"   5 minutes ago       Up 5 minutes        6379/tcp                                                                                 awx_redis
2a26d2655542        memcached:alpine                        "docker-entrypoint.s…"   5 minutes ago       Up 5 minutes        11211/tcp                                                                                awx_memcached
ee76bd7a7655        postgres:10                             "docker-entrypoint.s…"   5 minutes ago       Up 5 minutes        5432/tcp                                                                                 awx_postgres

```

**Comment by ryanpetrello** [comment]
You're using the 9.3.0 image, but you're deploying from `devel` (which is where the just-merged redis implementation lives - not that your install has a `redis` container, rather than a `rabbitmq` container).

see: #6392 

Your options are:

Clone the 9.3.0 tag instead of devel: https://github.com/ansible/awx/releases/tag/9.3.0
If you want to use devel, use your own image until the next major version of AWX is released (10.0.0): https://github.com/ansible/awx/blob/devel/INSTALL.md#official-vs-building-images
